gpiozero/spi_devices.py

```diff
@@ -72,10 +111,10 @@ class AnalogInputDevice(SPIDevice):
     .. _analog to digital converters: https://en.wikipedia.org/wiki/Analog-to-digital_converter
     """
 
-    def __init__(self, bits=None, **spi_args):
-        if bits is None:
-            raise InputDeviceError('you must specify the bit resolution of the device')
+    def __init__(self, bits, **spi_args):
         self._bits = bits
+        self._min_value = -(2 ** bits)
+        self._range = 2 ** (bits + 1) - 1
```

Comment:
Okay, so for full-differential devices the range is `2 ** (bits + 1) - 1`, but presumably for pseudo-differential ADCs (or non-differential ADCs) the range is still `2 ** bits - 1` ?

Or is the code simply over-simplifying, in order to keep code-reuse to a maximum? (and specialisation to a minimum)